src/widgets/helper/ResizingTextEdit.cpp

```diff
@@ -270,7 +275,53 @@ bool ResizingTextEdit::canInsertFromMimeData(const QMimeData *source) const
 
 void ResizingTextEdit::insertFromMimeData(const QMimeData *source)
 {
-    if (!source->hasImage())
+    if (source->hasImage())
+    {
+#if 0
+        static QUrl url("http://localhost:7494/upload?password=xd");
+        // default port and password for nuuls' filehost.
+#else
+        static QUrl url("https://i.nuuls.com/upload");
+        // production site
+#endif
+
+        QImage image = qvariant_cast<QImage>(source->imageData());
+
+        const char *boundary = "thisistheboudaryasd";
+        QByteArray dataToSend;
+        dataToSend.insert(0, "--");
+        dataToSend.append(boundary);
+        dataToSend.append(
+            "\r\n"
+            "Content-Disposition: form-data; name=\"attachment\"; "
+            "filename=\"control_v.png\"\r\n"
+            "Content-Type: image/png\r\n"
+            "\r\n");
+
+        QByteArray imageData;
+        QBuffer buf(&imageData);
```

Comment:
is the pasted image always of type image/png, or do we possibly have to do a check for pasted mime type and upload as jpeg/gif/whatever?

Reply:
It is saved as a png from QImage. It cannot be another mime type.

Reply:
does that mean posting animated GIFs is not supported at this time?

Reply:
yeah sadly